bpaotu/bpaotu/tabular.py

```diff
@@ -31,32 +31,63 @@ def str_none_blank(v):
                 return ''
             return str(v)
 
-        csv_fd = io.StringIO()
-        w = csv.writer(csv_fd)
-        fields = []
-        heading = []
-        write_fns = []
+        headings = {}
+        write_fns = {}
         for column in SampleContext.__table__.columns:
-            fields.append(column.name)
             units = getattr(column, 'units', None)
             if column.name == 'id':
-                heading.append('BPA ID')
+                headings[column.name] = "BPA ID"
             else:
                 title = display_name(column.name)
                 if units:
                     title += ' [%s]' % units
-                heading.append(title)
+                headings[column.name] = title
 
             if column.name == 'id':
-                write_fns.append(format_bpa_id)
+                write_fns[column.name] = format_bpa_id
             elif hasattr(column, "ontology_class"):
-                write_fns.append(make_ontology_export(column.ontology_class))
+                write_fns[column.name] = make_ontology_export(column.ontology_class)
             else:
-                write_fns.append(str_none_blank)
-        w.writerow(heading)
-        for sample in samples:
-            w.writerow(f(getattr(sample, field)) for (field, f) in zip(fields, write_fns))
-        return csv_fd.getvalue()
+                write_fns[column.name] = str_none_blank
+
+    def get_context_value(sample, field):
+        return write_fns[field](getattr(sample, field))
+
+    # Issue: https://github.com/BioplatformsAustralia/bpaotu/issues/79
+    # Find all non-empty fields(except 'BPA ID')
```

Comment:
I don't think we need to reference the issue, it's not going to be terribly interesting down the track. It's definitely worth leaving in the 'find all non-empty fields' comment to explain what we're doing and why.

Reply:
removed isssu ref